spec/models/orders/order_spec.rb

```diff
@@ -41,10 +49,110 @@
 
     context 'with autodetect_studies_projects set to false' do
       let(:autodetect_studies_projects) { false }
+      let(:autodetect_studies) { nil }
+      let(:autodetect_projects) { nil }
       let(:aliquots) { create_list :tagged_aliquot, 2, study: study, project: project }
 
       it { is_expected.not_to be_valid }
     end
+
+    describe 'combining autodetection attributes' do
+      let(:aliquots) { create_list :tagged_aliquot, 2 }
+
+      context 'with all attributes set to true' do
+        let(:autodetect_studies_projects) { true }
+        let(:autodetect_studies) { true }
+        let(:autodetect_projects) { true }
+
+        it 'fails validation' do
+          expect(order).not_to be_valid
+          expect(
+            order.errors.full_messages.one? { |message| message.include? 'autodetect_studies_projects' }
+          ).to be true
+        end
+
+        it 'uses the defaults' do
```

Comment:
This is because it actually uses the value before the validation runs - see the following line:
`before_validation :set_study_from_aliquots, unless: :cross_study_allowed, if: :autodetect_studies?`

So I've got to make a decision about the behaviour even when there's a clash, because it won't have run the validation.

Happy to change it - what would you do here? (Or can just add a comment)

Reply:
I would just add a comment in the test about it I think.